src/matrixrtc/MatrixRTCSession.ts

```diff
@@ -232,30 +270,39 @@ export class MatrixRTCSession extends TypedEventEmitter<MatrixRTCSessionEvent, M
      * desired.
      * This method will return immediately and the session will be joined in the background.
      *
-     * @param activeFoci - The list of foci to set as currently active in the call member event
-     * @param manageMediaKeys - If true, generate and share a a media key for this participant,
-     *                          and emit MatrixRTCSessionEvent.EncryptionKeyChanged when
-     *                          media keys for other participants become available.
+     * @param fociActive - The object representing the active focus. (This depends on the focus type.)
+     * @param fociPreferred - The list of preferred foci this member proposes to use/knows/has access to.
+     *                        For the livekit case this is a list of foci generated from the homeserver well-known, the current rtc session,
+     *                        or optionally other room members homeserver well known.
+     * @param joinConfig - Additional configuration for the joined session.
      */
-    public joinRoomSession(activeFoci: Focus[], manageMediaKeys?: boolean): void {
+    public async joinRoomSession(
+        fociPreferred: Focus[],
+        fociActive?: Focus,
+        joinConfig?: JoinSessionConfig,
+    ): Promise<void> {
```

Comment:
Now that there's no more `await`, this can go back to being synchronous.
```suggestion
    public joinRoomSession(
        fociPreferred: Focus[],
        fociActive?: Focus,
        joinConfig?: JoinSessionConfig,
    ): void {
```